lib/std/mem.zig

```diff
@@ -1829,6 +1829,49 @@ test "replace" {
     testing.expect(eql(u8, output[0..], "Favor reading  over writing ."));
 }
 
+/// Replace all occurences of `needle` with `replacement`.
+pub fn replaceElem(comptime T: type, slice: []T, needle: T, replacement: T) void {
```

Comment:
If we follow the `std.mem` naming this should be `replaceScalar` and take an `output: []T` parameter (slice becomes `const`).
Specifying the same in/out buffers gets you the in-place replacement you want.

Reply:
fixed